<div style="text-align:left;font-size:2em"><span style="font-weight:bolder;font-size:1.25em">SP2273 | Learning Portfolio</span><br><br><span style="font-weight:bold;color:darkred">Using Jupyter (Good), Exercises</span></div>

# Exercise 1

In [4]:
while True:
    print('*', end='')

****************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************

KeyboardInterrupt: 

# Exercise 2

In [2]:
print = 'PRINT'

print('hello world')

# Excercise 3

In [11]:
squares = [i**2 for i in range(10)] 
print

<function print(*args, sep=' ', end='\n', file=None, flush=False)>

In [10]:
squares = list()
for i in range(10):
    squares.append(i**2)
print

<function print(*args, sep=' ', end='\n', file=None, flush=False)>

In [16]:
%timeit squares = [i**2 for i in range(10)]

427 ns ± 2.31 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)


In [17]:
%timeit squares = list()
for i in range(10):
    squares.append(i**2)

38.9 ns ± 0.131 ns per loop (mean ± std. dev. of 7 runs, 10,000,000 loops each)


For the first one, the average execution time of the code is 437 nanoseconds, while the second one is 38.9 nanoseconds. Hence, the second code is faster.

The first code was run 7 times and executed 1 000 000 times to calculate the standard deviation (2.31 ns per loop).

The second code was run 7 times and executed 10 000 000 times to calculate the stndard deciation (0.131 ns per loop). This shows that the second code's result can be more accurate due to the larger sample size.

